### ISSUE TITLE
CI error: System.Net.Quic.QuicException: The connection timed out from inactivity

### ISSUE THREAD

**Issue by steveharter** [dotnet/runtime#91757]
## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=640353
Build error leg or test failing: System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_MaxResponseHeadersLength_Http3.LargeSingleHeader_ThrowsException

## Error Message

Multiple `System.Net.Quic.QuicException : The connection timed out from inactivity.` in System.Net.Quic tests on "smaller" platforms, e.g. Alpine3.15 arm 32

```
 System.Net.Quic.QuicException : The connection timed out from inactivity.
      Stack Trace:
        /_/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs(547,0): at System.Net.Quic.QuicConnection.HandleConnectionEvent(QUIC_CONNECTION_EVENT& connectionEvent)
        /_/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs(581,0): at System.Net.Quic.QuicConnection.NativeCallback(QUIC_HANDLE* connection, Void* context, QUIC_CONNECTION_EVENT* connectionEvent)
        --- End of stack trace from previous location ---
        /_/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ValueTaskSource.cs(176,0): at System.Net.Quic.ValueTaskSource.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)
        /_/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs(324,0): at System.Net.Quic.QuicConnection.FinishConnectAsync(QuicClientConnectionOptions options, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs(72,0): at System.Net.Quic.QuicConnection.<ConnectAsync>g__StartConnectAsync|2_0(QuicClientConnectionOptions options, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs(77,0): at System.Net.Quic.QuicConnection.<ConnectAsync>g__StartConnectAsync|2_0(QuicClientConnectionOptions options, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicTestBase.cs(214,0): at System.Net.Quic.Tests.QuicTestBase.CreateConnectedQuicConnection(QuicClientConnectionOptions clientOptions, QuicListener listener)
        /_/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicTestBase.cs(245,0): at System.Net.Quic.Tests.QuicTestBase.CreateConnectedQuicConnection(QuicClientConnectionOptions clientOptions, QuicListener listener)
        /_/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicConnectionTests.cs(315,0): at System.Net.Quic.Tests.QuicConnectionTests.AcceptStreamAsync_ConnectionDisposed_Throws()
```

```json
{
  "ErrorPattern": "System.Net.Quic.QuicException ?: The connection timed out from inactivity",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=640353
**Error message validated:** `[System.Net.Quic.QuicException ?: The connection timed out from inactivity`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 4/16/2024 11:07:05 AM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[745864](https://dev.azure.com/dnceng-public/public/_build/results?buildId=745864)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.WriteTests](https://dev.azure.com/dnceng-public/public/_build/results?buildId=745864&view=ms.vss-test-web.build-test-results-tab&runId=18850132&resultId=210860)|dotnet/runtime#104680|
|[743164](https://dev.azure.com/dnceng-public/public/_build/results?buildId=743164)|dotnet/runtime|[System.Net.Quic.Tests.QuicStreamTests.WriteAsync_LocalAbort_Throws](https://dev.azure.com/dnceng-public/public/_build/results?buildId=743164&view=ms.vss-test-web.build-test-results-tab&runId=18773360&resultId=223759)|dotnet/runtime#101624|
|[742676](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742676)|dotnet/runtime|[System.Net.Quic.Functional.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742676&view=ms.vss-test-web.build-test-results-tab&runId=18758312&resultId=220356)|dotnet/runtime#104856|
|[742110](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742110)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocksProxyTest_Http2.TestLoopbackAsync](https://dev.azure.com/dnceng-public/public/_build/results?buildId=742110&view=ms.vss-test-web.build-test-results-tab&runId=18738674&resultId=212290)||
|[741273](https://dev.azure.com/dnceng-public/public/_build/results?buildId=741273)|dotnet/runtime|[System.Net.Quic.Tests.QuicStreamTests.WriteAsync_LocalAbort_Throws](https://dev.azure.com/dnceng-public/public/_build/results?buildId=741273&view=ms.vss-test-web.build-test-results-tab&runId=18715736&resultId=223750)|dotnet/runtime#101624|
|[740929](https://dev.azure.com/dnceng-public/public/_build/results?buildId=740929)|dotnet/runtime|[System.Net.Quic.Tests.QuicStreamConformanceTests.ReadWrite_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=740929&view=ms.vss-test-web.build-test-results-tab&runId=18705038&resultId=223720)|dotnet/runtime#101624|
|[739937](https://dev.azure.com/dnceng-public/public/_build/results?buildId=739937)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandlerTest_AutoRedirect.GetAsync_AllowAutoRedirectTrue_RetainsOriginalFragmentIfAppropriate](https://dev.azure.com/dnceng-public/public/_build/results?buildId=739937&view=ms.vss-test-web.build-test-results-tab&runId=18666962&resultId=212656)||
|[738918](https://dev.azure.com/dnceng-public/public/_build/results?buildId=738918)|dotnet/runtime|[System.Net.Quic.Tests.QuicStreamTests.WriteAsync_LocalAbort_Throws](https://dev.azure.com/dnceng-public/public/_build/results?buildId=738918&view=ms.vss-test-web.build-test-results-tab&runId=18641532&resultId=223695)|dotnet/runtime#101624|
|[737434](https://dev.azure.com/dnceng-public/public/_build/results?buildId=737434)|dotnet/runtime|[System.Net.Quic.Tests.QuicStreamTests.WriteAsync_LocalAbort_Throws](https://dev.azure.com/dnceng-public/public/_build/results?buildId=737434&view=ms.vss-test-web.build-test-results-tab&runId=18597218&resultId=223678)|dotnet/runtime#101624|
|[736217](https://dev.azure.com/dnceng-public/public/_build/results?buildId=736217)|dotnet/runtime|[System.Net.Quic.Tests.MsQuicTests.WriteTests](https://dev.azure.com/dnceng-public/public/_build/results?buildId=736217&view=ms.vss-test-web.build-test-results-tab&runId=18557616&resultId=219794)|dotnet/runtime#104684|
|[735197](https://dev.azure.com/dnceng-public/public/_build/results?buildId=735197)|dotnet/runtime|[System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_Http11_Cancellation_Test.GetAsync_CancelDuringResponseHeadersReceived_TaskCanceledQuickly](https://dev.azure.com/dnceng-public/public/_build/results?buildId=735197&view=ms.vss-test-web.build-test-results-tab&runId=18527874&resultId=212842)||
|[734813](https://dev.azure.com/dnceng-public/public/_build/results?buildId=734813)|dotnet/runtime|[System.Net.Http.Functional.Tests.HttpClientHandlerTest_Http3.ServerClosesOutboundControlStream_ClientClosesConnection](https://dev.azure.com/dnceng-public/public/_build/results?buildId=734813&view=ms.vss-test-web.build-test-results-tab&runId=18507324&resultId=213068)|dotnet/runtime#104630|
|[713739](https://dev.azure.com/dnceng-public/public/_build/results?buildId=713739)|dotnet/runtime|[System.Net.Quic.Tests.QuicStreamConformanceTests.ReadWrite_MessagesSmallerThanReadBuffer_Success](https://dev.azure.com/dnceng-public/public/_build/results?buildId=713739&view=ms.vss-test-web.build-test-results-tab&runId=17854038&resultId=188389)|dotnet/runtime#103646|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|7|13|

**Comment by liveans** [comment]
This test was activated for the #74896 issue. The failures are different but they're coming from the same issue, I'm investigating that.

**Comment by lewing** [comment]
This is causing a lot of failures, is there a timeline for a fix?

**Comment by liveans** [comment]
Looks like recent failures are coming from different tests and from `Alpine.315.Arm32.Open` mostly.

cc: @ManickaP

**Comment by ManickaP** [comment]
I've separated the disabled test into a new issue and I'm keeping this for the QUIC timeout

**Comment by karelz** [comment]
@JulieLeeMSFT yes, it popped in our weekly Networking CI report as well. We are digging into it -- see pending PR #94683 and merged PR #94609

**Comment by CarnaViire** [comment]
Added "disabled-test" tag, as the issue was just temporarily mitigated by pinning older docker tag (with older msquic revision) in https://github.com/dotnet/runtime/pull/94683 and https://github.com/dotnet/runtime/pull/94609. Tracking external issue is https://github.com/microsoft/msquic/issues/3958

**Comment by CarnaViire** [comment]
Also, trying to pin the image to a specific msquic commit instead of main in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/933, to be able to start tracking docker image changes again (by reverting https://github.com/dotnet/runtime/pull/94683 and https://github.com/dotnet/runtime/pull/94609).

It doesn't yet seem that the external issue would be fixed soon, so we cannot block everyone else on this (e.g. https://github.com/dotnet/dnceng/issues/1423)

**Comment by CarnaViire** [comment]
@karelz this exact problem should not affect CI currently (because of https://github.com/dotnet/runtime/pull/94683 and https://github.com/dotnet/runtime/pull/94609)... is it still appearing?

**Comment by karelz** [comment]
This specific issue had a lot of hits in last month. Perhaps it was just stale data showing there.

**Comment by MihaZupan** [comment]
This was tracking the same pattern as #94507 with the only difference being `QuicException: The` vs `QuicException : The`.
I've updated the pattern here to match both.

**Comment by CarnaViire** [comment]
> Tracking external issue is https://github.com/microsoft/msquic/issues/3958

MsQuic's issue should be fixed in v2.4 release (which is not out yet)
